### PR TITLE
Support angular 4

### PR DESCRIPTION
**Pull request by dxvladislavvolkov**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,18 +16,18 @@
...
-    "@angular/common": "^2.2.1",
-    "@angular/forms": "^2.2.1"
+    "@angular/core": "^2.2.1 || ^4.0.0-rc.1",
+    "@angular/common": "^2.2.1 || ^4.0.0-rc.1",
+    "@angular/forms": "^2.2.1 || ^4.0.0-rc.1"
   },
   "devDependencies": {
     "devextreme": "~16.2.4",
-    "jquery": "^2.0.0 || ^3.0.0",
+    "jquery": "^2.0.0 || ~3.1.1",
```
Comment (kvet) [inline package.json]: Remove this unnecessary dev dependency

File: package.json [hunk 2]
```diff
@@ -16,18 +16,18 @@
   "author": "Developer Express Inc.",
   "license": "MIT",
   "dependencies": {
-    "@types/jquery": "^2.0.0 || ^3.0.0",
-    "jquery": "^2.0.0 || ^3.0.0"
+    "@types/jquery": "^2.0.0 || ~3.1.1",
+    "jquery": "^2.0.0 || ~3.1.1"
```
Comment (kvet) [inline package.json]: Seems that "2.0.0 - 3.2" is more suitable version specification

Comment (GoshaFighten) [inline package.json]: Why do we need these changes?

Comment (dxvladislavvolkov) [inline package.json]: These changes fixed issue [#406](https://github.com/DevExpress/devextreme-angular/issues/406)

Comment (GoshaFighten) [inline package.json]: Yes, but we will be stuck to jQuery 3.1.1. jQuery is going to fix the issue in [$.holdReady is broken in 3.2.0](https://github.com/jquery/jquery/issues/3573), plus we are going fix it in our code in the [DevExtreme fails with the "$.holdReady is not a function" error when jQuery 3.2.0 is used](https://www.devexpress.com/Support/Center/Question/Details/T494217) bug report.

File: package.json (hunk 1 above)
Comment (Aden-git) [inline package.json]: @kvet 
I'm not sure, that this is unnecessary devDependency. We using it at least in tests.

**Pull request merged**